experiments/billable/src/worker.tsx

```diff
@@ -1,155 +1,54 @@
-      // (from experimentation, it is not in node_modules/.vite). This means that if you were to subsequently
-      // change the code to _not_ have Prisma used after the initial request, the WASM will still be cached and
-      // the request will not hang. This makes this issue particularly hard to debug.
-      await db.$queryRaw`SELECT 1`;
-
-      const url = new URL(request.url);
-
-      if (url.pathname === '/test/db') {
-        console.log('## test db')
-        const r = await db.$queryRaw`SELECT 1`
-        console.log('## test db result', r)
-        return new Response(JSON.stringify(r), { status: 200 })
-      }
-
-      let ctx: Awaited<ReturnType<typeof getContext>> = {};
-      let session: Awaited<ReturnType<typeof getSession>> | undefined;
-      try {
-        session = await getSession(request, env);
-        ctx = await getContext(session);
-      } catch (e) {
-        console.error("Error getting session", e);
-      }
-
-      const isRSCRequest = url.searchParams.has("__rsc");
-      const isRSCActionHandler = url.searchParams.has("__rsc_action_id");
-      let actionResult: any;
-      if (isRSCActionHandler) {
-        actionResult = await rscActionHandler(request, ctx); // maybe we should include params and ctx in the action handler?
-      }
-
-      const renderPage = async (Page: any, props = {}) => {
-        const rscPayloadStream = renderToRscStream({
-          node: <Page {...props} />,
-          actionResult: actionResult,
-        });
-        if (isRSCRequest) {
-          return new Response(rscPayloadStream, {
-            headers: { "content-type": "text/x-component; charset=utf-8" },
-          });
-        }
-        const [rscPayloadStream1, rscPayloadStream2] = rscPayloadStream.tee();
-
-        const htmlStream = await transformRscToHtmlStream({
-          stream: rscPayloadStream1,
-          Parent: App,
-        });
+};
 
-        const html = htmlStream.pipeThrough(
-          injectRSCPayload(rscPayloadStream2),
-        );
-        return new Response(html, {
-          headers: { "content-type": "text/html" },
+const routes = [
+  index([
+    function ({ ctx }) {
+      if (ctx.user) {
+        return new Response(null, {
+          status: 302,
+          headers: { Location: link('/invoice/list') },
         });
-      };
-
-      const response = await router.handle({
-        request,
-        ctx,
-        env,
-        renderPage,
-      });
-      return response;
-    } catch (e) {
-      if (e instanceof ErrorResponse) {
-        return new Response(e.message, { status: e.code });
       }
-
-      console.error("Unhandled error", e);
-      throw e;
-    }
-  },
-};
+    },
+    HomePage,
+  ]),
+  ...prefix("/user", authRoutes),
+  ...prefix("/invoice", invoiceRoutes),
+]
+
+export default defineApp<ReturnType<typeof getContext>>({
+  Head,
+  getContext,
+  routes,
+})
```

Comment:
@peterp This is how the worker entry point looks now.

Totally don't mind this changing entirely, was just more straightforward to do for a first iteration.

Idea is that this just returns cloudflare worker handlers (e.g. `fetch`), so that the user can wrap this to do their own things. For example, the user can do:

```ts
const app = defineApp(...)

export default {
  fetch(request, env) {
    if (/* condition */) {
      return app.fetch(request, env)
   } else {
      return /* something else */
   }
  }
}
```